internal/icinga2/client_api.go

```diff
@@ -309,9 +320,19 @@ func (client *Client) connectEventStream(esTypes []string) (io.ReadCloser, error
 		queueNameRndBuff := make([]byte, 16)
 		_, _ = rand.Read(queueNameRndBuff)
 
-		reqBody, err := json.Marshal(map[string]any{
+		reqBody := new(bytes.Buffer)
+		encoder := json.NewEncoder(reqBody)
+		// Since our encoded JSON isn't going to be used somewhere in a browser, we've to disable this feature
+		// so that our filter expressions such as ‘&’ don't get replaced by some UTF-8 code points.
+		encoder.SetEscapeHTML(false)
+
+		err := encoder.Encode(map[string]any{
 			"queue": fmt.Sprintf("icinga-notifications-%x", queueNameRndBuff),
 			"types": esTypes,
+			"filter": fmt.Sprintf(
+				"(event.type!=%q && event.type!=%q) || event.object_type==%q || event.object_type==%q",
+				typeObjectCreated, typeObjectDeleted, "Host", "Service",
+			),
```

Comment:
> Object created for all Icinga 2 objects.

This is a quote from the Icinga 2 event stream docs for the `ObjectCreated` event and means that this event is triggered for every single object of Icinga 2 of any type, but I don't really care about any other types than `Host` and `Service`. Therefore, I want to check if `event.type` is not `ObjectCreated` and not `ObjectDeleted` (this means that the event type could be any other type listed in `esTypes`), otherwise (this means that `event.type` is either `ObjectCreated` or `ObjectDeleted`), I want to additionally filter on the `object_type` and match only on the `Host` or `Service` types.

Reply:
> Something like the following should also work as a filter:

Did you test it? Otherwise you should have already noticed that this doesn't work.

> I think this would be more readable if it was a little more verbose if expression.

This is a simple API filter expression, I'm not sure what exactly you find it to be *unreadable*, and apart from that I just didn't want to use the literal `ObjectCreated` while there is a constant `typeObjectCreated` for it, and **no**, you can't just use `event.object_type == "Host" || event.object_type == "Service"` or `(event.type == "ObjectCreated" || event.type == "ObjectDeleted")` as a filter value, you have to use quoted values `"\"FOO\""`.

Reply:
> > Something like the following should also work as a filter:
> 
> Did you test it? Otherwise you should have already noticed that this doesn't work.

I didn't test it within the code here, I just verified in `icinga2 console` that `if` expressions indeed evaluate to a value so I'd expect it to work here as well:

```
Icinga 2 (version: v2.14.2)
Type $help to view available commands.
<1> => if (true) { 23 } else { 42 }
23.000000
<2> => if (false) { 23 } else { 42 }
42.000000
```

> > I think this would be more readable if it was a little more verbose if expression.
> 
> This is a simple API filter expression, I'm not sure what exactly you find it to be _unreadable_, and apart from that I just didn't want to use the literal `ObjectCreated` while there is a constant `typeObjectCreated` for it

That comment was primarily about the structure of the condition. Isn't the purpose of the filter to perform a check for some event types and leave the other ones unfiltered? I think such an `if` would convey this in a more straight-forward way.

> and **no**, you can't just use `event.object_type == "Host" || event.object_type == "Service"` or `(event.type == "ObjectCreated" || event.type == "ObjectDeleted")` as a filter value, you have to use quoted values `"\"FOO\""`.

Yes, you can, you can alternatively enclose string literals in ``` `` ``` and then don't need to escape double quotes.